zebra-state/src/service/chain_tip.rs

```diff
@@ -387,6 +391,38 @@ impl ChainTip for LatestChainTip {
         self.with_chain_tip_block(|block| block.transaction_hashes.clone())
             .unwrap_or_else(|| Arc::new([]))
     }
+
+    /// Returns when the state tip changes.
+    ///
+    /// Marks the state tip as seen.
+    #[instrument(skip(self))]
+    fn best_tip_changed(&mut self) -> BestTipChanged {
```

Comment:
Nitpick: it seems simpler to use `std::pin::Pin<Box<dyn Future<Output = Result<(), watch::error::RecvError>> + Send + '_>>` (or add the async-trait crate and wait for the feature to hit rust stable https://blog.rust-lang.org/inside-rust/2022/11/17/async-fn-in-trait-nightly.html#path-to-stabilization)

Reply:
I don't really mind what we do here, but here are a few reasons I did it this way:

This is the way the standard library does it, see `FutureExt` for some similar future combinator types. So we know it will continue to work, even on nightly.

I didn't want to add an extra proc macro dependency just for one type. (It adds to compilation time, because it has to be compiled every time we compile Zebra.)

I tried spelling out the underlying type everywhere, and it didn't quite work, because it still carried over some of the input type generics in `MockChainTip::changed()`. So I had to use two levels of `BestTipChanged` to fully erase the type there.

I would also prefer to not repeat the exact type everywhere, so it's easy to change in the one place. And it's easier to read. If we didn't need to erase the extra generics, we could just use a type alias. But I don't think that would work here.

Reply:
I'm happy to take a suggestion PR for this, but I'd like to avoid conflicts with the tests I need to write.

Reply:
> This is the way the standard library does it

> I didn't want to add an extra proc macro dependency just for one type. (It adds to compilation time, because it has to be compiled every time we compile Zebra.)

That makes sense, it would be simplifying marking the value as seen and I don't think we need to do that here anyways, so this is good.